test/cpp/api/transformer.cpp

```diff
@@ -150,37 +158,20 @@ TEST_F(TransformerTest, TransformerEncoderLayer_CUDA) {
 }
 
 void transformer_decoder_layer_test_helper(bool is_cuda){
-  int64_t d_model = 4;
-  int64_t nhead = 2;
-  int64_t dim_feedforward = 16;
-  double dropout = 0.0;
-
-  TransformerDecoderLayer model = TransformerDecoderLayer(
-    TransformerDecoderLayerOptions(d_model, nhead)
-    .dim_feedforward(dim_feedforward).dropout(dropout));
-  if (is_cuda) {
-    model->to(torch::kCUDA);
-  }
 
   torch::Device device = is_cuda ? torch::kCUDA : torch::kCPU;
   torch::TensorOptions tensor_options = torch::TensorOptions()
     .dtype(torch::kFloat32).device(device);
 
-  //set constant weights of the model
-  {
-    torch::NoGradGuard guard;
-    for (const auto& parameter : model->parameters()) {
-      at::Tensor x = parameter.data();
-      auto sz = x.view(-1).size(0);
-      auto shape = x.sizes();
-      x = torch::cos(torch::arange(0, sz, tensor_options).view(shape));
-      parameter.data().copy_(x);
-    }
-  }
+  TransformerDecoderLayer model = get_a_test_layer<
+    TransformerDecoderLayer,
+    TransformerDecoderLayerOptions>(tensor_options);
 
   // deterministic input
-  at::Tensor decoder_input = torch::tensor({{{20, 30, 40, 50}}}, tensor_options);
-  at::Tensor memory_input = torch::tensor({{{60, 70, 80, 90}}}, tensor_options);
+  at::Tensor decoder_input = torch::tensor({{{20, 30, 40, 50}}},
```

Comment:
Please use torch::Tensor instead of at::Tensor for the following lines where ever at::Tensor is there.